op-chain-ops/cmd/withdrawals/main.go

```diff
@@ -32,64 +43,244 @@ func main() {
+			bn, err := withdrawals.WaitForFinalizationPeriod(context.Background(), l1Client, optimismPortalAddr, transitionBlockNumber)
+			if err != nil {
+				return err
+			}
+			header, err := l2Client.HeaderByNumber(context.Background(), new(big.Int).SetUint64(bn))
+			if err != nil {
+				return err
+			}
+			proof, err := gclient.GetProof(context.Background(), predeploys.L2ToL1MessagePasserAddr, []string{slot.String()}, header.Number)
+			if err != nil {
+				return err
+			}
+			if len(proof.StorageProof) != 1 {
+				return errors.New("invalid amount of storage proofs")
+			}
+			trieNodes := make([][]byte, len(proof.StorageProof[0].Proof))
+			for i, s := range proof.StorageProof[0].Proof {
+				trieNodes[i] = common.FromHex(s)
+			}
+
+			portal, err := bindings.NewOptimismPortal(optimismPortalAddr, l1Client)
+			if err != nil {
+				return err
+			}
+
+			l2OracleAddr, err := portal.L2ORACLE(&bind.CallOpts{})
+			oracle, err := bindings.NewL2OutputOracleCaller(l2OracleAddr, l1Client)
+			if err != nil {
+				return nil
+			}
+
+			l2OutputIndex, err := oracle.GetL2OutputIndexAfter(&bind.CallOpts{}, header.Number)
+
+			if ctx.String("private-key") == "" {
+				return errors.New("No private key to transact with")
+			}
+			privateKey, err := crypto.HexToECDSA(strings.TrimPrefix(ctx.String("private-key"), "0x"))
+			if err != nil {
+				return err
+			}
+
+			opts, err := bind.NewKeyedTransactorWithChainID(privateKey, l1ChainID)
+			if err != nil {
+				return err
+			}
+
+			tx, err := portal.ProveWithdrawalTransaction(
+				opts,
+				bindings.TypesWithdrawalTransaction{
+					Nonce:    withdrawal.Nonce,
+					Sender:   *withdrawal.Sender,
+					Target:   *withdrawal.Target,
+					Value:    withdrawal.Value,
+					GasLimit: withdrawal.GasLimit,
+					Data:     withdrawal.Data,
+				},
+				l2OutputIndex,
+				bindings.TypesOutputRootProof{
+					Version:                  [32]byte{},
+					StateRoot:                header.Root,
+					MessagePasserStorageRoot: proof.StorageHash,
+					LatestBlockhash:          header.Hash(),
+				},
+				trieNodes,
+			)
+
+			if err != nil {
+				return err
+			}
+
+			receipt, err := bind.WaitMined(context.Background(), l1Client, tx)
+			if err != nil {
+				return err
+			}
+			if receipt.Status != types.ReceiptStatusSuccessful {
+				return errors.New("withdrawal proof unsuccessful")
+			}
+
+			log.Info("Withdrawal proven", "tx-hash", tx.Hash(), "withdrawal-hash", hash)
+
+			block, err := l1Client.BlockByHash(context.Background(), receipt.BlockHash)
+			if err != nil {
+				return err
+			}
+
+			// Get the finalization period
+			finalizationPeriod, err := portal.FINALIZATIONPERIODSECONDS(&bind.CallOpts{})
+			if err != nil {
+				return err
+			}
+
+			future := block.Time() + finalizationPeriod.Uint64() + 1
+			var result bool
+			// TODO: double check this is the correct RPC
+			err = l1RpcClient.Call(&result, "hardhat_setTimestamp", future)
```

Comment:
We can remove this now

Reply:
The finalization period should be 2 seconds, we should be able to just wait for it